### ISSUE TITLE
Worker scale up issue "Waiting for kubelet,kube-proxy to start"

### ISSUE THREAD

**Issue by boucherv** [juju-solutions/bundle-canonical-kubernetes#519]
Hello,

After a Kubernetes deployment, when I do a worker scale up sometime the Worker still waiting for kubelet and kubeproxy service ("Waiting for kubelet,kube-proxy to start").

It's appear that the charm don't create some config file for those services (kube-proxyconfig, args service files etc)

Here it's the file with all log generated by "cdk-field-agent" [results-2018-03-23-11-32-07.tar.gz](https://github.com/juju-solutions/bundle-canonical-kubernetes/files/1842318/results-2018-03-23-11-32-07.tar.gz)

Before the scale up, I do a scale down so it's possible that one cdk component don't really clean his config and have an issue with an IP already known.

Thanks in advance,
Valentin

**Comment by hansbogert** [comment]
Seems to be a hostname issue. If I change the hostname of worker node, it starts working

**Comment by hansbogert** [comment]
@boucherv do you happen to use MAAS?

**Comment by boucherv** [comment]
@hansbogert Yes! The name is the same after a scale down/scale up because we keep the machine in MAAS

**Comment by hyperbolic2346** [comment]
It looks like cdk-addons.apply isn't being run on that worker, which means the snap isn't being configured. Not sure how this ties into the hostname being the same yet. I assume these machines are wiped before being recycled?

**Comment by hansbogert** [comment]
@hyperbolic2346 I tried both with "fresh"/wiped nodes as well as with MAAS nodes that were hosting multiple services and re-used them for the k8s-worker. Both of these types of nodes failed if the node was already used for the k8s-worker service -- at least that's seems to be the distinguishing factor. I do want to stipulate my previous comment:

> Seems to be related to at least the hostname. If I change the hostname of worker node, it starts working

**Comment by hyperbolic2346** [comment]
> Seems to be related to at least the hostname. If I change the hostname of worker node, it starts working

Yes, I think this is the largest clue so far. I just wanted to make sure it was the hostname and not something latent on the filesystem. Thanks for the clarification.

**Comment by fl8858** [comment]
I ran into the similar issue, after powering off and then powering on the worker. It ran ok before. After restarting all the servers, the master status became "Waiting for kube-system pods to start", and the worker status "Waiting for kubelet to start.". The system runs on MAAS.

**Comment by dniel** [comment]
I got a very similar problem today https://discourse.jujucharms.com/t/waiting-for-kubelet-kube-proxy-to-start/2036 that I still haven't managed to fix.

**Comment by SDAdham** [comment]
I am experiencing this issue too today

**Comment by SDAdham** [comment]
> Seems to be related to at least the hostname. If I change the hostname of worker node, it starts working

Hi @hansbogert : can you please share more details about changing the hostname here? i.e from what to what?

**Comment by SDAdham** [comment]
@hyperbolic2346 can you please help?

**Comment by SDAdham** [comment]
my environment:
```
kubernetes-master      1.15.3   active           2  kubernetes-master      jujucharms  724  ubuntu
kubernetes-worker      1.15.4   waiting          2  kubernetes-worker      jujucharms  571  ubuntu  exposed
```
$ journalctl -u snap.kubelet.daemon.service
Sep 17 09:18:28 eager-grub kubelet.daemon[14384]: I0917 09:18:28.778113   14384 client.go:75] Connecting to docker on unix:///var/run/docker.sock
Sep 17 09:18:28 eager-grub kubelet.daemon[14384]: I0917 09:18:28.778233   14384 client.go:104] Start docker client with request timeout=2m0s
Sep 17 09:18:28 eager-grub kubelet.daemon[14384]: F0917 09:18:28.779493   14384 server.go:273] failed to run Kubelet: failed to create kubelet: failed to get docker version: Cannot connect to the Docker daemon at unix:///var/run/docker.s
Sep 17 09:18:28 eager-grub systemd[1]: snap.kubelet.daemon.service: Main process exited, code=exited, status=255/n/a
Sep 17 09:18:28 eager-grub systemd[1]: snap.kubelet.daemon.service: Failed with result 'exit-code'.
Sep 17 09:18:29 eager-grub systemd[1]: snap.kubelet.daemon.service: Service hold-off time over, scheduling restart.
Sep 17 09:18:29 eager-grub systemd[1]: snap.kubelet.daemon.service: Scheduled restart job, restart counter is at 3.
Sep 17 09:18:29 eager-grub systemd[1]: Stopped Service for snap application kubelet.daemon.
Sep 17 09:18:29 eager-grub systemd[1]: Started Service for snap application kubelet.daemon.
```

**Comment by hyperbolic2346** [comment]
This could be https://bugs.launchpad.net/charm-kubernetes-worker/+bug/1839704

**Comment by hansbogert** [comment]
@SDAdham as mentioned earlier, the bug I and OP was hitting was fixed. Are you at the least using MAAS/Juju? Since that was part of the underlying problem. Don't fixate too much on the exact text of "Waiting for kubelet,kube-proxy to start", it can happen for a lot of reasons.
